core/templates/pages/exploration-player-page/layout-directives/learner-view-info.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    statsReportingService = TestBed.inject(StatsReportingService);
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    urlService = TestBed.inject(UrlService);
+    storyViewerBackendApiService = TestBed.inject(StoryViewerBackendApiService);
+  });
+
+  afterEach(() => {
+    componentInstance.ngOnDestory();
+  });
+
+  it('should initialize', fakeAsync(() => {
+    let explorationId = 'expId';
+    let explorationTitle = 'Exploration Title';
+    let topicUrl = 'topic_url';
+
+    spyOn(urlService, 'getPathname').and.returnValue('/explore/');
+    spyOn(contextService, 'getExplorationId').and.returnValue(explorationId);
+    spyOn(readOnlyExplorationBackendApiService, 'fetchExplorationAsync')
+      .and.returnValue(Promise.resolve({
+        exploration: {
+          title: explorationTitle
+        }
+      } as FetchExplorationBackendResponse));
+
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(componentInstance, 'getTopicUrl').and.returnValue(topicUrl);
+    spyOn(urlService, 'getTopicUrlFragmentFromLearnerUrl').and.returnValue('');
+    spyOn(urlService, 'getClassroomUrlFragmentFromLearnerUrl')
+      .and.returnValue('');
+    spyOn(urlService, 'getStoryUrlFragmentFromLearnerUrl').and.returnValue('');
+    spyOn(storyViewerBackendApiService, 'fetchStoryDataAsync').and.returnValue(
+      Promise.resolve(new StoryPlaythrough('', [], '', '', '', '')));
+    spyOn(statsReportingService, 'setTopicName');
+    spyOn(siteAnalyticsService, 'registerCuratedLessonStarted');
+
+    componentInstance.ngOnInit();
+    tick();
+    tick();
+    expect(urlService.getPathname).toHaveBeenCalled();
+    expect(contextService.getExplorationId).toHaveBeenCalled();
+    expect(readOnlyExplorationBackendApiService.fetchExplorationAsync)
+      .toHaveBeenCalled();
+    expect(urlService.getExplorationVersionFromUrl).toHaveBeenCalled();
+    expect(componentInstance.getTopicUrl).toHaveBeenCalled();
+    expect(urlService.getTopicUrlFragmentFromLearnerUrl).toHaveBeenCalled();
+    expect(urlService.getStoryUrlFragmentFromLearnerUrl).toHaveBeenCalled();
+    expect(storyViewerBackendApiService.fetchStoryDataAsync).toHaveBeenCalled();
+    expect(statsReportingService.setTopicName).toHaveBeenCalled();
+    expect(siteAnalyticsService.registerCuratedLessonStarted)
+      .toHaveBeenCalled();
+  }));
+
+  it('should get topic url', () => {
+    let topicUrl = 'topic_url';
+    spyOn(urlService, 'getTopicUrlFragmentFromLearnerUrl').and.returnValue(
+      'topic_url_fragment');
+    spyOn(urlService, 'getClassroomUrlFragmentFromLearnerUrl').and.returnValue(
+      'classroom_url_fragment');
+    spyOn(urlInterpolationService, 'interpolateUrl').and.returnValue(topicUrl);
+    expect(componentInstance.getTopicUrl()).toEqual(topicUrl);
```

Comment:
Split the code inside the `it` block into applicable sections (Setup, Pre-check, Action, Post-check) by adding a new line between the sections

Reply:
done